### PR TITLE
Java Grep RCR

### PR DESCRIPTION
**Pull request by ashwinrishipj**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,43 @@ target/
 *.logs
```
Comment (jarviscanada) [inline .gitignore:6]: target/
*.class
*.jar
*.war
*.ear
*.logs
*.iml
.idea/
.eclipse

File: core_java/grep/out/grep [hunk 2]
```diff
@@ -0,0 +1,146 @@
+  Romeo, son to Montague.
+  Mercutio, kinsman to the Prince and friend to Romeo.
+  Benvolio, nephew to Montague, and friend to Romeo
```
Comment (jarviscanada) [inline core_java/grep/out/grep]: git rm -r core_java/grep/out

File: core_java/grep/out/grep.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
+Library of the Future and Shakespeare CDROMS.  Project Gutenberg
+Shakespeare
+in the presentation of The Complete Works of William Shakespeare
+The Complete Works of William Shakespeare 
+The Library of the Future Complete Works of William Shakespeare 
+Please give them our thanks for their Shakespeare cooperation!
+by William Shakespeare
```
Comment (jarviscanada) [inline core_java/grep/out/grep.txt]: git rm

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		writeToFile(matchedLines);
+	}
+
+	@Override
+	public List<File> listFiles(String directory) {
+		try {
+			filesList = Files.list(Paths.get(directory)).filter(Files::isRegularFile).map(Path::toFile)
+					.collect(Collectors.toList());
+		} catch (Exception e) {
+			JavaGrepImp.logger.error("Reading files error:" + e);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: RTE

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+		lines = new ArrayList<String>();
+		try {
+			scanner = new Scanner(inputFile);
+			while (scanner.hasNextLine()) {
+				String line = scanner.nextLine();
+				lines.add(line);
+			}
+			scanner.close();
+		} catch (FileNotFoundException e) {
+			JavaGrepImp.logger.error("Reading files error:" + e);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: remove logger

**Review by jarviscanada — APPROVED** [review]
LGTM

File: core_java/grep/out/grep (hunk 2 above)
Comment (ashwinrishipj) [reply core_java/grep/out/grep]: removed.

File: core_java/grep/out/grep.txt (hunk 3 above)
Comment (ashwinrishipj) [reply core_java/grep/out/grep.txt]: removed.

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java (hunk 4 above)
Comment (ashwinrishipj) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: updated.

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java (hunk 5 above)
Comment (ashwinrishipj) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: removed and updated.

**Pull request merged**